lab_1/main.py

```diff
@@ -30,7 +53,19 @@ def calculate_frequencies(tokens: list) -> dict or None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
-    pass
+
+    freq_dict = {}
+    if not isinstance(tokens, list):
+        return None
+    for i in tokens:
+        if not isinstance(i, str):
+            return None
+        for _ in tokens:
```

Comment:
`_` is usually used for variables that you don't need
You definitely need this one